avm/src/vm/opcode.rs

```diff
@@ -3167,22 +3159,24 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
     Box::pin(async move {
       hand_mem.dupe(args[0], args[0]); // Make sure there's no pointers involved
       let fractal = hand_mem.read_fractal(args[0]);
-      let conn_id = fractal.read_fixed(3);
-      let responses = Arc::clone(&HTTP_RESPONSES);
       let mut hm = HandlerMemory::new(None, 1);
       HandlerMemory::transfer(&hand_mem, args[0], &mut hm, CLOSURE_ARG_MEM_START);
       let res_out = hm.read_fractal(CLOSURE_ARG_MEM_START);
       for i in 0..res_out.len() {
         hm.register_from_fractal(i as i64, &res_out, i);
       }
-      responses.insert(conn_id, hm);
-      drop(responses);
-      // TODO: Add a second synchronization tool to return a valid Result status, for now, just
-      // return success
-      hand_mem.init_fractal(args[2]);
-      hand_mem.push_fixed(args[2], 0i64);
-      hand_mem.push_fractal(args[2], HandlerMemory::str_to_fractal("ok"));
-      hand_mem
+      unsafe {
+        // get watch tx from the raw ptr
+        let tx_raw_ptr = fractal.read_fixed(3) as *const Sender<Arc<HandlerMemory>>;
```

Comment:
is this always guaranteed to be at block 3?

Reply:
unless we change https://github.com/alantech/alan/pull/449/commits/2fbb1ce0c77c79fb2bd49b199a70cc6cd12ec59f#diff-aa0e5a81bac19bdacb5f06f8d0cefe2516b53e02227ac03e058f2230d02f7ba2R3055, yes